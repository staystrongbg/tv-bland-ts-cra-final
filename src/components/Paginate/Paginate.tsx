type Props = {
  pageNum: number[];
  paginate: (page: number) => void;
};

const Paginate: React.FC<Props> = ({ pageNum, paginate }) => {
  const PAGINATION_STYLE = {
    width: '100%',
    gap: '5px',
  };

  const LI_STYLE = {
    cursor: 'pointer',
    border: '1px solid #1e1e1e',
    boxShadow: '1px 1px rgba(0,0,0,0.6)',
    width: '30px',
    height: '30px',
    color: '#f5f5f5',
    background: '#101010',
    borderRadius: '3px',
  };
  return (
    <ul className={`${PAGINATION_STYLE} flex-center`} style={PAGINATION_STYLE}>
      {pageNum?.map((page, idx) => (
        <li
          style={LI_STYLE}
          key={idx}
          className='flex-center'
          onClick={() => paginate(page)}
        >
          {page}
        </li>
      ))}
    </ul>
  );
};

export default Paginate;
