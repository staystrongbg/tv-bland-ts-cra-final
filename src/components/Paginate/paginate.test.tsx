import React from 'react';
import renderer from 'react-test-renderer';
import Paginate from './Paginate';

it('render app', () => {
  let pageNum = [1, 2],
    paginate = (page = 2) => page;

  const render = renderer
    .create(<Paginate pageNum={pageNum} paginate={paginate} />)
    .toJSON();
  expect(render).toMatchSnapshot();
});
