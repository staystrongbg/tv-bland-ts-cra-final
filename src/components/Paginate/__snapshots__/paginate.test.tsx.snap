// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app 1`] = `
<ul
  className="[object Object] flex-center"
  style={
    Object {
      "gap": "5px",
      "width": "100%",
    }
  }
>
  <li
    className="flex-center"
    onClick={[Function]}
    style={
      Object {
        "background": "#101010",
        "border": "1px solid #1e1e1e",
        "borderRadius": "3px",
        "boxShadow": "1px 1px rgba(0,0,0,0.6)",
        "color": "#f5f5f5",
        "cursor": "pointer",
        "height": "30px",
        "width": "30px",
      }
    }
  >
    1
  </li>
  <li
    className="flex-center"
    onClick={[Function]}
    style={
      Object {
        "background": "#101010",
        "border": "1px solid #1e1e1e",
        "borderRadius": "3px",
        "boxShadow": "1px 1px rgba(0,0,0,0.6)",
        "color": "#f5f5f5",
        "cursor": "pointer",
        "height": "30px",
        "width": "30px",
      }
    }
  >
    2
  </li>
</ul>
`;
